src/lib/game/game.ts

```diff
@@ -0,0 +1,88 @@
+import { get, writable } from 'svelte/store';
+import { Timer } from './timer';
+
+export class Game {
+	private readonly _goal = 10;
+	private _timer = new Timer();
+	private _scoreA = writable(0);
+	private _scoreB = writable(0);
+	private _started = writable(false);
+	private _paused = writable(false);
+
+	get duration() {
+		return this._timer.duration;
+	}
+
+	get scoreA() {
+		return this._scoreA;
+	}
+
+	get scoreB() {
+		return this._scoreB;
+	}
+
+	get started() {
+		return this._started;
+	}
+
+	get paused() {
+		return this._paused;
+	}
+
+	start() {
+		if (get(this._started)) {
+			//throw error
+		}
+
+		console.log('starting game');
+		this._timer.start();
+		this._started.set(true);
+		//Create GameStartedEvent
+	}
+
+	pause() {
+		if (!this._timer.stop()) {
+			//throw error
+		}
+
+		console.log('pausing game');
+		this._timer.stop();
+		this._paused.set(true);
+		//Create GamePausedEvent
+	}
+
+	resume() {
+		if (get(this._started)) {
+			//throw error
+		}
+
+		console.log('resuming game');
+		this._timer.start();
+		this._paused.set(false);
+		//Create GameResumedEvent
+	}
+
+	scoreGoal(team: Team) {
```

Comment:
When getting the scores there are two seperate methods. Feels weird that setting and getting is different in this way.
Not sure which one I prefer

Reply:
I agree. I think perhaps one method for each?

Reply:
Yeah I think so too. Either way it's easy to change later if needed, so nice to just choose one for now

Reply:
Just need a clean way to tell the Counter which team it belongs to

Reply:
This is a problem with moving the logic away from the component and into game.
Could game just own the score value? It could then react to changes if needed.
But the counter could own the function increasing the value. The game doesn't really need it

Reply:
This is a good point. I'll give an alternative solution that is probably cleaner.

Reply:
Never mind, I'm not sure the way I was thinking of is cleaner. I would like to pair-program this if possible.